apiserver/uniter/uniter_test.go

```diff
@@ -2755,11 +2763,391 @@ func (s *uniterNetworkConfigSuite) TestNetworkConfigForImplicitlyBoundEndpoint(c
+		Bindings: []string{"db", "admin-api", "db-client"},
+	}
+	// For the relation "wordpress:db mysql:server" we expect to see only
+	// ifaces ibn the "internal" space, where the "db" endpoint itself
+	// is bound to.
+	expectedConfigWithRelationName := params.NetworkInfoResult{
+		Info: []params.NetworkInfo{
+			{
+				MACAddress:    "00:11:22:33:44:50",
+				InterfaceName: "eth0.100",
+				Addresses: []params.InterfaceAddress{
+					{Address: "10.0.0.10", CIDR: "10.0.0.0/24"},
+				},
+			},
+			{
+				MACAddress:    "00:11:22:33:44:52",
+				InterfaceName: "eth1.100",
+				Addresses: []params.InterfaceAddress{
+					{Address: "10.0.0.11", CIDR: "10.0.0.0/24"},
+				},
+			},
+		},
+	}
+	// For the "admin-api" extra-binding we expect to see only interfaces from
+	// the "public" space.
+	expectedConfigWithExtraBindingName := params.NetworkInfoResult{
+		Info: []params.NetworkInfo{
+			{
+				MACAddress:    "00:11:22:33:44:50",
+				InterfaceName: "eth0",
+				Addresses: []params.InterfaceAddress{
+					{Address: "8.8.8.10", CIDR: "8.8.0.0/16"},
+				},
+			},
+			{
+				MACAddress:    "00:11:22:33:44:51",
+				InterfaceName: "eth1",
+				Addresses: []params.InterfaceAddress{
+					{Address: "8.8.4.10", CIDR: "8.8.0.0/16"},
+					{Address: "8.8.4.11", CIDR: "8.8.0.0/16"},
+				},
+			},
+		},
+	}
+
+	// For the "db-client" extra-binding we expect to see interfaces from default
+	// "wp-default" space
+	expectedConfigWithDefaultSpace := params.NetworkInfoResult{
+		Info: []params.NetworkInfo{
+			{
+				MACAddress:    "00:11:22:33:44:52",
+				InterfaceName: "eth2",
+				Addresses: []params.InterfaceAddress{
+					{Address: "100.64.0.10", CIDR: "100.64.0.0/16"},
+				},
+			},
+		},
+	}
+
+	result, err := s.base.uniter.NetworkInfo(args)
+	c.Assert(err, jc.ErrorIsNil)
+	c.Assert(result, jc.DeepEquals, params.NetworkInfoResults{
+		Results: map[string]params.NetworkInfoResult{
+			"db":        expectedConfigWithRelationName,
+			"admin-api": expectedConfigWithExtraBindingName,
+			"db-client": expectedConfigWithDefaultSpace,
+		},
+	})
+}
+
+func (s *uniterNetworkInfoSuite) TestNetworkInfoL2Binding(c *gc.C) {
+	c.Skip("L2 not supported yet")
+	s.addRelationAndAssertInScope(c)
+
+	args := params.NetworkInfoParams{
+		Unit:     s.base.wordpressUnit.Tag().String(),
+		Bindings: []string{"foo-bar"},
+	}
+
+	expectedInfo := params.NetworkInfoResult{
+		Info: []params.NetworkInfo{
+			{
+				MACAddress:    "00:11:22:33:44:50",
+				InterfaceName: "eth2",
+			},
+		},
+	}
+
+	result, err := s.base.uniter.NetworkInfo(args)
+	c.Assert(err, jc.ErrorIsNil)
+	c.Assert(result, jc.DeepEquals, params.NetworkInfoResults{
+		Results: map[string]params.NetworkInfoResult{
+			"foo-bar": expectedInfo,
+		},
+	})
+}
+
+func (s *uniterNetworkInfoSuite) TestNetworkInfoForImplicitlyBoundEndpoint(c *gc.C) {
+	// Since wordpressUnit has explicit binding for "db", switch the API to
+	// mysqlUnit and check "mysql:server" uses the machine preferred private
+	// address.
```

Comment:
Yes, that is a fallback that was there and I don't want to change the behaviour.